CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKCommandBuffer.java

```diff
@@ -146,8 +180,294 @@ public void FinishRecord(){
+
+		bindDescriptorSetsCmd(pipelinyLayout, descriptorSets,
+				VK_PIPELINE_BIND_POINT_COMPUTE);
+	}
+//	
+//	public void bindGraphicsDescriptorSetsCmd(long pipelinyLayout, long[] descriptorSets){
+//		
+//		bindDescriptorSetsCmd(pipelinyLayout, descriptorSets,
+//				VK_PIPELINE_BIND_POINT_GRAPHICS);
+//	}
+//	
+	private void bindDescriptorSetsCmd(long pipelinyLayout, long[] descriptorSets,
+			int pipelineBindPoint){
+		
+		vkCmdBindDescriptorSets(vkCommandBuffer, pipelineBindPoint,
+				pipelinyLayout, 0, descriptorSets, null);
+	}
+	
+//	public void clearColorImageCmd(long image, int imageLayout){
+//		
+//		VkImageSubresourceRange subresourceRange = VkImageSubresourceRange.calloc()
+//				.aspectMask(VK_IMAGE_ASPECT_COLOR_BIT)
+//				.baseMipLevel(0)
+//				.levelCount(1)
+//				.baseArrayLayer(0)
+//				.layerCount(1);
+//		
+//		vkCmdClearColorImage(vkCommandBuffer, image, imageLayout,
+//				CVKUtils.getClearColorValue(), subresourceRange);
+//	}
+//	
+//	public void drawIndexedCmd(int indexCount){
+//		
+//		vkCmdDrawIndexed(vkCommandBuffer, indexCount, 1, 0, 0, 0);
+//	}
+//	
+//	public void drawCmd(int vertexCount){
+//		
+//		vkCmdDraw(vkCommandBuffer, vertexCount, 1, 0, 0);
+//	}
+//	
+//	public void dispatchCmd(int groupCountX, int groupCountY, int groupCountZ){
+//		
+//		vkCmdDispatch(vkCommandBuffer, groupCountX, groupCountY, groupCountZ);
+//	}
+//	
+//	public void copyBufferCmd(long srcBuffer, long dstBuffer,
+//								    long srcOffset, long dstOffset,
+//								    long size){
+//		
+//		VkBufferCopy.Buffer copyRegion = VkBufferCopy.calloc(1)
+//					.srcOffset(srcOffset)
+//					.dstOffset(dstOffset)
+//					.size(size);
+//		
+//		vkCmdCopyBuffer(vkCommandBuffer, srcBuffer, dstBuffer, copyRegion);
+//	}
+//	
+//	public void copyBufferToImageCmd(long srcBuffer, long dstImage, int width, int height, int depth){
+//		
+//		VkBufferImageCopy.Buffer copyRegion = VkBufferImageCopy.calloc(1)
+//					.bufferOffset(0)
+//					.bufferRowLength(0)
+//					.bufferImageHeight(0);
+//		
+//		VkImageSubresourceLayers subresource = VkImageSubresourceLayers.calloc()
+//					.aspectMask(VK_IMAGE_ASPECT_COLOR_BIT)
+//					.mipLevel(0)
+//					.baseArrayLayer(0)
+//					.layerCount(1);
+//		
+//		VkExtent3D extent = VkExtent3D.calloc()
+//					.width(width)
+//					.height(height)
+//					.depth(depth);
+//		
+//		VkOffset3D offset = VkOffset3D.calloc()
+//					.x(0)
+//					.y(0)
+//					.z(0);
+//		
+//		copyRegion.imageSubresource(subresource);
+//		copyRegion.imageExtent(extent);
+//		copyRegion.imageOffset(offset);
+//	
+//		vkCmdCopyBufferToImage(vkCommandBuffer, srcBuffer, dstImage,
+//				VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, copyRegion);
+//	}
+//	
+	public void pipelineImageMemoryBarrierCmd(long image, int oldLayout, int newLayout,
+			int srcAccessMask, int dstAccessMask, int srcStageMask, int dstStageMask,
+			int baseMipLevel, int mipLevelCount){
+		
+            // TODO change to callocStack?
```

Comment:
Yes I think so.

Reply:
Will change with the CommandBuffer refactor